### DISCUSSION TITLE
How to call GetUserEffectivePermissionsAsync on library without wiping model data?

### DISCUSSION THREAD

**Discussion by wizofaus** [pnp/pnpcore#1416] [Q&A] [answered]
I have code where I've performed a complex query on a list to get all the data I need about it. I need the model data to remain available, but I also need to call GetUserEffectivePermissionsAsync( ) for the list. However currently this wipes the key model data (e.g. list.Fields.AsRequested()).  While I *can* issue a second call to PnPContext.Web.GetListByIdAsync( ) this means another network call that shouldn't be necessary as I already know the list ID, and don't need any further information about the list.
Basically what I want is the ability to *clone* the list to a new object that has the same ID and PnPContext, which I can then perform new queries on without blowing model data away on the original list.
I've scoured the source code and I'm really struggling to see how this is possible.
Thanks
Dylan

**Comment by jansenbe** [comment]
@wizofaus : I think there are some options, let me explore and report back

**Reply by jansenbe** [reply, marked as answer]
@wizofaus : I've added a fix for this (see https://github.com/pnp/pnpcore/commit/1280e570c68818da866639291578648f803d7f0b). Will be part of next nightly

**Comment by wizofaus** [comment]
Ok, thanks, though I still feel it's useful to be able to obtain a clean List instance for a given Guid without having to issue a network call.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
